src/testing/tree_construction.rs

```diff
@@ -267,11 +141,10 @@ impl Test {
             document = DocumentBuilder::new_document();
         };
 
-        // Create a new parser
         let options = Html5ParserOptions { scripting_enabled };
 
         let mut chars = CharIterator::new();
-        chars.read_from_str(self.data(), None);
+        chars.read_from_str(self.test.spec.data.as_str(), None);
```

Comment:
We should probably keep `test.spec` private and expose `self.test.as_str()`, which delegates to `test.spec.data.as_str()`.

(no need to block on this detail).

Reply:
we do the same with `document` from the spec_data, so this makes sense